src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,150 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode;
+        if (head == null) {
+            newNode = new Node<>(null, value, null);
+            head = tail = newNode;
+        } else {
+            newNode = new Node<>(this.tail, value, null);
+            newNode.prev = tail;
+            tail.next = newNode;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndex(index, size);
+        Node<T> newNode;
+        if (head == null || index == size) {
+            add(value);
+            return;
+        } else if (index == 0) {
+            newNode = new Node<>(null, value, head);
+            head = newNode;
+            newNode.next.prev = newNode;
+        } else {
+            Node<T> current = getNode(index - 1, size);
+            newNode = new Node<>(current.prev, value, current.next);
+            newNode.next.prev = newNode;
+            current.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (list == null) {
+            return;
+        }
+        for (T value : list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNode(index, size - 1).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> node = getNode(index, size - 1);
+        T oldvalue = node.value;
+        node.value = value;
+        return oldvalue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> removed = getNode(index, size - 1);
+        unlink(removed);
+        return removed.value;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> temp = head;
```

Comment:
better to give variable understandable names. Like `current`, `nodeToDelete` etc